test/test-udp-send-unreachable.c

```diff
@@ -138,12 +155,44 @@ TEST_IMPL(udp_send_unreachable) {
                   send_cb);
   ASSERT(r == 0);
 
+  r = uv_udp_init(uv_default_loop(), &client2);
+  ASSERT(r == 0);
+
+  r = uv_udp_bind(&client2, (const struct sockaddr*) &addr3, UV_UDP_RECVERR);
+  ASSERT(r == 0);
+
+  if (can_recverr) {
+    r = uv_udp_recv_start(&client2, alloc_cb, recv_cb);
+    ASSERT(r == 0);
+
+    /* client sends "PING", then "PANG" */
+    buf = uv_buf_init("PING", 4);
+
+    r = uv_udp_send(&req3,
+		    &client2,
+		    &buf,
+		    1,
+		    (const struct sockaddr*) &addr,
+		    send_cb_recverr);
+    ASSERT(r == 0);
+
+    buf = uv_buf_init("PANG", 4);
+
+    r = uv_udp_send(&req4,
+		    &client2,
+		    &buf,
+		    1,
+		    (const struct sockaddr*) &addr,
+		    send_cb_recverr);
+    ASSERT(r == 0);
+  }
+
   uv_run(uv_default_loop(), UV_RUN_DEFAULT);
 
-  ASSERT(send_cb_called == 2);
-  ASSERT(recv_cb_called == alloc_cb_called);
-  ASSERT(timer_cb_called == 1);
-  ASSERT(close_cb_called == 2);
+  ASSERT_EQ(send_cb_called, (long)(can_recverr ? 4 : 2));
```

Comment:
Is the cast necessary? And if so, why `long` instead of `int`? `send_cb_called` is of type int.

Reply:
Changed the cast to `(int)` in 99ca2862049961e0156f2fbd359074842aa8152b, but I actually think there was a warning compiling the code, otherwise I wouldn't add the cast :).